src/readfish/plugins/utils.py

```diff
@@ -792,3 +797,109 @@ def get_action(self, channel: int) -> Optional[Action]:
         :return: The last action sent for the channel, or None if no action has been sent.
         """
         return self.last_actions.get(channel, None)
+
+
+@attrs.define
+class DuplexTracker:
+    """
+    Wrapper class to keep track the alignment location of the latest read seen on a channel,
+    and previous decision made, tracking whether we made a duplex override on the last read
+    Specifically, we store a list of tuples of any target contig names and strands that were aligned to,
+    keyed to channel number and the previous decision for a read made on that channel.
+    The decision should only be updated when a read has been finalised and should not be seen again,
+    i.e a Stop receiving or Unblock has been sent to MinKNOW
+    No maps are specified as (*, *)
+    """
+
+    previous_alignments: Dict[int, list[tuple[str, Strand]]] = attrs.Factory(dict)
+    previous_decision: Dict[int, Decision] = attrs.Factory(dict)
```

Comment:
I originally was using the AlignmentProtocol - I switched because we don't need to keep anything but the `contig` and `Strand`, and I didn't want all the excess hanging around, just nice fast and immmutable tuples

I think for now we can leave it like this? If we ever do more complex duplex we should change to use the `Alignment` type?

Reply:
Happy to go with that, maybe the signature on the `possible_duplex` should account for the strand type. Perhaps a future note/comment so we don't forget?

Reply:
Added note for future coders!